hummingbot/connector/exchange/bittrex/bittrex_auth.py

```diff
@@ -1,56 +1,49 @@
-import time
-import hmac
 import hashlib
+import hmac
 import urllib
-from typing import Dict, Any, Tuple
+import uuid
+from typing import Any, Dict
 
-import ujson
+from hummingbot.connector.time_synchronizer import TimeSynchronizer
+from hummingbot.core.web_assistant.auth import AuthBase
+from hummingbot.core.web_assistant.connections.data_types import RESTRequest, WSRequest
 
 
-class BittrexAuth:
-    def __init__(self, api_key: str, secret_key: str):
+class BittrexAuth(AuthBase):
+    def __init__(self, api_key: str, secret_key: str, time_provider: TimeSynchronizer):
         self.api_key = api_key
         self.secret_key = secret_key
-
-    def generate_auth_dict(
-        self,
-        http_method: str,
-        url: str,
-        params: Dict[str, Any] = None,
-        body: Dict[str, Any] = None,
-        subaccount_id: str = "",
-    ) -> Dict[str, any]:
-        """
-        Generates the url and the valid signature to authenticate with the API endpoint.
-        :param http_method: String representing the HTTP method in use ['GET', 'POST', 'DELETE'].
-        :param url: String representing the API endpoint.
-        :param params: Dictionary of url parameters to be included in the api request. USED ONLY IN SOME CASES
-        :param body: Dictionary representing the values in a request body.
-        :param subaccount_id: String value of subaccount id.
-        :return: Dictionary containing the final 'params' and its corresponding 'signature'.
-        """
-
-        # Appends params the url
-        def append_params_to_url(url: str, params: Dict[str, any] = {}) -> str:
-            if params:
-                param_str = urllib.parse.urlencode(params)
-                return f"{url}?{param_str}"
-            return url
-
-        def construct_content_hash(body: Dict[str, any] = {}) -> Tuple[str, bytes]:
-            json_byte: bytes = "".encode()
-            if body:
-                json_byte = ujson.dumps(body).encode()
-                return hashlib.sha512(json_byte).hexdigest(), json_byte
-            return hashlib.sha512(json_byte).hexdigest(), json_byte
-
-        timestamp = str(int(time.time() * 1000))
-        url = append_params_to_url(url, params)
-        content_hash, content_bytes = construct_content_hash(body)
-        content_to_sign = "".join([timestamp, url, http_method, content_hash, subaccount_id])
+        self.time_provider = time_provider
+
+    async def rest_authenticate(self, request: RESTRequest) -> RESTRequest:
+        headers = {}
+        if request.headers:
+            headers.update(request.headers)
+        headers.update(self.generate_REST_auth_params(request=request))
+        request.headers = headers
+        return request
+
+    async def ws_authenticate(self, request: WSRequest) -> WSRequest:
+        return request  # pass-through
+
+    @staticmethod
+    def construct_content_hash(body) -> str:
```

Comment:
Missing type hints (please also check in the rest of the classes to make sure there are no type hints missing)

Reply:
Added type hints